From 7e8f6eff32048c35113dd883b554e94f44bd33a0 Mon Sep 17 00:00:00 2001
Message-Id: <7e8f6eff32048c35113dd883b554e94f44bd33a0.1414143271.git.chang-joon.lee@intel.com>
In-Reply-To: <f393e797ccaaf66a25ee3d7c4e6136083a6effeb.1414143271.git.chang-joon.lee@intel.com>
References: <f393e797ccaaf66a25ee3d7c4e6136083a6effeb.1414143271.git.chang-joon.lee@intel.com>
From: Tom O'Rourke <Tom.O'Rourke@intel.com>
Date: Tue, 21 Oct 2014 16:05:16 -0700
Subject: [PATCH 23/26] Revert "MUST_REBASE [VPG]: drm/i915: Signal curd in
 intel_gen6_powersave_work"

This reverts commit 34db415755307ad6e106ba595fea8e8cd3994ed2.

Conflicts:

	drivers/gpu/drm/i915/i915_drv.h

Change-Id: I5ea1a29e7ae986fcda9629fd4fa2fa81926307af
For: GMIN-3534
Signed-off-by: Tom O'Rourke <Tom.O'Rourke@intel.com>
---
 drivers/gpu/drm/i915/i915_drv.h   |    5 -----
 drivers/gpu/drm/i915/i915_sysfs.c |   42 -------------------------------------
 drivers/gpu/drm/i915/intel_pm.c   |    5 -----
 3 files changed, 52 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_drv.h b/drivers/gpu/drm/i915/i915_drv.h
index 88bea72..9588b6a 100644
--- a/drivers/gpu/drm/i915/i915_drv.h
+++ b/drivers/gpu/drm/i915/i915_drv.h
@@ -1608,11 +1608,6 @@ struct drm_i915_private {
 	/* Display functions */
 	struct drm_i915_display_funcs display;
 
-	/* curd information */
-	struct {
-		struct pid *pid;
-	} curd;
-
 	/* DPST information */
 	struct {
 		struct pid *pid;
diff --git a/drivers/gpu/drm/i915/i915_sysfs.c b/drivers/gpu/drm/i915/i915_sysfs.c
index c21cf2d..bf05b81 100644
--- a/drivers/gpu/drm/i915/i915_sysfs.c
+++ b/drivers/gpu/drm/i915/i915_sysfs.c
@@ -105,46 +105,6 @@ forcewake_store(struct device *kdev, struct device_attribute *attr,
 }
 
 static ssize_t
-show_curd(struct device *kdev, struct device_attribute *attr, char *buf)
-{
-	struct drm_minor *minor = dev_to_drm_minor(kdev);
-	struct drm_device *dev = minor->dev;
-	struct drm_i915_private *dev_priv = dev->dev_private;
-	ssize_t count = 0;
-
-	count = snprintf(buf, PAGE_SIZE, "%d\n", pid_nr(dev_priv->curd.pid));
-
-	return count;
-}
-
-static ssize_t
-curd_store(struct device *kdev, struct device_attribute *attr,
-		const char *buf, size_t count)
-{
-	struct drm_minor *minor = dev_to_drm_minor(kdev);
-	struct drm_device *dev = minor->dev;
-	struct drm_i915_private *dev_priv = dev->dev_private;
-	struct pid *p;
-	u32 val;
-	ssize_t ret;
-
-	ret = kstrtou32(buf, 0, &val);
-	if (ret)
-		return ret;
-
-	if (1 == val) {
-		p = get_task_pid(current, PIDTYPE_PID);
-		put_pid(dev_priv->curd.pid);
-		dev_priv->curd.pid = p;
-	} else if (0 == val) {
-		put_pid(dev_priv->curd.pid);
-		dev_priv->curd.pid = NULL;
-	}
-
-	return count;
-}
-
-static ssize_t
 show_rc6_mask(struct device *kdev, struct device_attribute *attr, char *buf)
 {
 	struct drm_minor *dminor = dev_to_drm_minor(kdev);
@@ -181,7 +141,6 @@ show_rc6pp_ms(struct device *kdev, struct device_attribute *attr, char *buf)
 
 static DEVICE_ATTR(forcewake, S_IRUSR | S_IWUSR, show_forcewake,
 		   forcewake_store);
-static DEVICE_ATTR(curd, S_IRUSR | S_IWUSR, show_curd, curd_store);
 static DEVICE_ATTR(rc6_enable, S_IRUGO, show_rc6_mask, NULL);
 static DEVICE_ATTR(rc6_residency_ms, S_IRUGO, show_rc6_ms, NULL);
 static DEVICE_ATTR(rc6p_residency_ms, S_IRUGO, show_rc6p_ms, NULL);
@@ -189,7 +148,6 @@ static DEVICE_ATTR(rc6pp_residency_ms, S_IRUGO, show_rc6pp_ms, NULL);
 
 static struct attribute *rc6_attrs[] = {
 	&dev_attr_forcewake.attr,
-	&dev_attr_curd.attr,
 	&dev_attr_rc6_enable.attr,
 	&dev_attr_rc6_residency_ms.attr,
 	&dev_attr_rc6p_residency_ms.attr,
diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index 8dc7b58..1ea9ff9 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -5800,11 +5800,6 @@ static void intel_gen6_powersave_work(struct work_struct *work)
 	dev_priv->rps.enabled = true;
 	mutex_unlock(&dev_priv->rps.hw_lock);
 
-	if (kill_pid_info(SIGHUP, SEND_SIG_FORCED, dev_priv->curd.pid)) {
-		put_pid(dev_priv->curd.pid);
-		dev_priv->curd.pid = NULL;
-	}
-
 	intel_runtime_pm_put(dev_priv);
 }
 
-- 
1.7.9.5

